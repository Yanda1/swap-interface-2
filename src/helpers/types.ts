import SOURCE_NETWORKS from '../data/sourceNetworks.json';
import DESTINATION_NETWORKS from '../data/destinationNetworks.json';
import { CONTRACT_ADDRESSES } from '../helpers';

export type Networks = typeof DESTINATION_NETWORKS;
export type Sources = typeof SOURCE_NETWORKS;
export type DestinationNetworks = keyof Networks;
export type ContractAdress = keyof typeof CONTRACT_ADDRESSES;

export type ApiAuthType = { access: string; is_kyced: boolean; refresh: string };

export type Price = { symbol: string; price: string };
export type Fee = { amount: number; currency: string; name?: string };
export type GraphType = false | { distance: number; path: string[] };

export type TransactionData = {
	blockNumber: number;
	header: {
		timestamp: number | undefined;
		symbol: string;
		scoin: string;
		fcoin: string;
		samt: string;
		net: string;
	};
	content:
		| {
				qty: string;
				price: string;
				timestamp: number;
				cexFee: string;
				withdrawFee: string;
				success: boolean;
		  }
		| null
		| 'none';
	gasFee: string;
	withdrawl: {
		amount: string;
		withdrawFee: string;
		url: string;
	} | null;
};

export type LocalStorageHistory = {
	[key in string]: { lastBlock: number | null; data: TransactionData[] };
};

export type TransactionHeaderSortValue =
	| undefined
	| 'fcoin'
	| 'scoin'
	| 'symbol'
	| 'timestamp'
	| 'samt'
	| 'net';

export type SelectProps = {
	name: string;
	value: TransactionHeaderSortValue;
	checked: boolean;
};

export type CostRequest = {
	data: string;
	id: string;
	customer: string;
	service: string;
	productId: string;
	validatorsList: string[];
	blockNumber: string;
	blockTimestamp: string;
	transactionHash: string;
};
