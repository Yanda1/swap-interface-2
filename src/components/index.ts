export * from './accordion/accordion';
export * from './button/button';
export * from './button/swapButton';
export * from './fees/fees';
export * from './header/header';
export * from './icon/icon';
export * from './modal/networkTokenModal';
export * from './modal/notificationsModal';
export * from './modal/portal';
export * from './modal/walletModal';
export * from './select/select';
export * from './selectList/selectList';
export * from './spinner/spinner';
export * from './switch/switch';
export * from './tabs/tabs';
export * from './tabs/tabContent';
export * from './textField/textField';
export * from './toast/toast';
export * from './wallet/wallet';
export * from './modal/kycL2Modal';
export * from './footer/footer';
