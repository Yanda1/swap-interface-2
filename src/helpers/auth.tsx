import { Buffer } from 'buffer';
import type { ApiAuthType } from '../helpers';
import { BASE_URL, BINANCE_PROD_URL, BINANCE_SCRIPT, BIZ_ENTRY_KEY, routes } from '../helpers';
import axios from 'axios';

const getMetamaskMessage = (nonce: string): string =>
	`0x${Buffer.from(
		'By signing this nonce: "' +
			nonce +
			'" you accept the terms and conditions available at https://cryptoyou.io/terms-of-use/',
		'utf8'
	).toString('hex')}`;

export const loadBinanceKycScript = (cb?: any) => {
	const existingId = document.getElementById('binance-kcy-script');

	if (!existingId) {
		const binanceSdkScript = document.createElement('script');
		binanceSdkScript.src = BINANCE_SCRIPT;
		binanceSdkScript.id = 'binance-kcy-script';
		document.body.appendChild(binanceSdkScript);

		binanceSdkScript.onload = () => {
			if (cb) cb();
		};
	}
	if (existingId && cb) cb();
};

export const makeBinanceKycCall = (authToken: string) => {
	// @ts-ignore
	const binanceKyc = new BinanceKyc({
		authToken,
		bizEntityKey: BIZ_ENTRY_KEY,
		apiHost: BINANCE_PROD_URL,
		onMessage: ({ typeCode }: any) => {
			if (typeCode === '102') {
				binanceKyc.switchVisible(true);
			}
		}
	});
};

export const getAuthTokensFromNonce = async (account: string, library: any) => {
	try {
		const res = await axios.request({
			url: `${BASE_URL}${routes.getNonce}${account}`
		});
		try {
			const msg = getMetamaskMessage(res.data.nonce);
			const signature = await library?.send('personal_sign', [account, msg]);
			try {
				const tokenRes = await axios.request({
					url: `${BASE_URL}${routes.auth}`,
					method: 'POST',
					data: { address: account, signature }
				});

				return tokenRes.data as ApiAuthType;
			} catch (err: any) {
				throw new Error(err);
			}
		} catch (err: any) {
			throw new Error(err);
		}
	} catch (err: any) {
		throw new Error(err);
	}
};
