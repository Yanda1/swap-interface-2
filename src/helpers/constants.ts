import { Mainnet, Moonbeam } from '@usedapp/core';

export const LOCAL_STORAGE_THEME = 'darkMode';
export const LOCAL_STORAGE_AUTH = 'auth';
export const LOCAL_STORAGE_HISTORY = 'history';

export const INITIAL_STORAGE = {
	access: '',
	refresh: '',
	account: '',
	isKyced: false
};

export const MESSAGE = {
	ignore: 'ignore',
	failed: 'Request failed'
};

export const BIZ_ENTRY_KEY = 'CRYPTOYOU';
export const BINANCE_PROD_URL = 'https://api.commonservice.io';
export const BINANCE_DEV_URL = 'https://dip-qacb.sdtaop.com';
export const BINANCE_SCRIPT =
	'https://static.saasexch.com/static/binance/static/kyc-ui/sdk/0.0.2/sdk.js';
export const BINANCE_PRICE_TICKER = 'https://www.binance.com/api/v3/ticker/price';
export const BINANCE_EXCHANGE_INFO = 'https://api.binance.com/api/v3/exchangeInfo';

export const BASE_URL = 'https://auth-app-ijjt6.ondigitalocean.app/';
export const MOONBEAM_URL = 'https://rpc.api.moonbeam.network';
export const ETHEREUM_URL = 'https://mainnet.infura.io/v3';

export const routes = {
	getNonce: 'auth?address=',
	auth: 'auth',
	kycToken: 'kyc/token',
	kycStatus: 'kyc/status',
	refresh: 'auth/refresh',
	transactionDetails: 'account/withdraw/details?id=',
	kycL2NaturalForm: 'kyc/l2-natural'
};

export const PROTOCOL_FEE = 0.002;
export const BINANCE_FEE = 0.002;
export const PROTOCOL_FEE_FACTOR = 1 / (1 - PROTOCOL_FEE);
export const FEE_CURRENCY = 'USDT';

export const ESTIMATED_NETWORK_TRANSACTION_GAS = 55_437;
export const SERVICE_ADDRESS = '0x47b211c494107C98160DF6dFA93D98175Ec2F27E';

export const BLOCK_CONTRACT_NUMBER = 2_075_594;
export const BLOCK_CHUNK_SIZE = 7_500;
export const WEI_TO_GLMR = 1 / 1_000_000_000_000_000_000;

export const CONTRACT_ADDRESSES = {
	1: '0xa9EB7218Fd8153c93aD1b4acf42330E7044E75A1',
	1284: '0xb8F18F75D5513F2fAA6477f0b54cC676eaedcAC4'
};

export const CHAINS = {
	// TODO: add types
	'1284': { name: 'GLMR', network: Moonbeam.chainId },
	'1': { name: 'ETH', network: Mainnet.chainId }
};

export const NETWORK_TO_ID = {
	GLMR: '1284',
	ETH: '1'
};

export const BLOCKS_AMOUNT = 30;

export const makeId = (length: number) => {
	let result = '';
	const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
	const charactersLength = characters.length;
	for (let i = 0; i < length; i++) {
		result += characters.charAt(Math.floor(Math.random() * charactersLength));
	}

	return result;
};
