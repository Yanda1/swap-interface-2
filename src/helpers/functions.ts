import { format, utcToZonedTime } from 'date-fns-tz';
import type { Theme } from './../styles';
import { DefaultSelectEnum } from '../helpers';

export const isLightTheme = (theme: Theme): boolean => theme.name === 'light';
const { timeZone: localTimeZone } = Intl.DateTimeFormat().resolvedOptions();
const timeZone =
	process.env.NODE_ENV === 'test' ? process.env.REACT_APP_TEST_TIMEZONE : localTimeZone;

export const isNetworkSelected = (network: string) =>
	network !== DefaultSelectEnum.NETWORK && network !== '';

export const isTokenSelected = (token: string) => token !== DefaultSelectEnum.TOKEN && token !== '';

export const isArrayType = (value: any) => typeof value === 'object' && Array.isArray(value);

type BeautifyNumbers = {
	n: string | number;
	digits?: number;
};

const trimZeros = (res: string): string =>
	res.slice(-1) === '0' && res.slice(-2, -1) !== '.' ? trimZeros(res.slice(0, -1)) : res;

export const beautifyNumbers = ({ n, digits = 8 }: BeautifyNumbers): string => {
	let res = '';
	if (!n) return '';
	if (typeof n === 'number') {
		res = n.toFixed(digits);
	} else {
		res = (+n).toFixed(digits);
	}

	return trimZeros(res);
};

export const hexToRgbA = (hex: string, alpha = '1'): string => {
	const [r, g, b] = hex.match(/\w\w/g)!.map((x) => parseInt(x, 16));

	return `rgba(${r},${g},${b},${alpha})`;
};

export const isSwapRejected = (status: string, errorMessage: any) =>
	status === 'Exception' && errorMessage === 'user rejected transaction';

export const isSwapFailed = (status: string) => status === 'Fail';

export const isSwapConfirmed = (status: string) => status === 'Success';

export const formatDate = (ts: number | undefined): string =>
	ts
		? format(utcToZonedTime(new Date(ts * 1000), timeZone as string), 'dd/MM/yyyy HH:mm:ss')
		: 'n/a';

export const findAndReplace = (array: string[], find: string, replace: string): string[] => {
	const result = [...array];
	if (result.includes(find)) {
		const index = result.indexOf(find);
		result[index] = replace;
	}

	return result;
};
