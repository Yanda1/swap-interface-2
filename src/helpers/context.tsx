import React, { createContext, ReactNode, useContext, useEffect, useReducer } from 'react';
import { BASE_URL } from './constants';
import type { ColorType, Theme } from '../styles';
import { darkTheme } from '../styles';
import axios from 'axios';

// TODO: should the enums be moved to the types.ts?
export enum VerificationEnum {
	ACCOUNT = 'SET_ACCOUNT_CONNECTED',
	NETWORK = 'SET_NETWORK_CONNECTED',
	USER = 'SET_USER_VERIFIED',
	ACCESS = 'SET_ACCESS_TOKEN',
	REFRESH = 'SET_REFRESH_TOKEN'
}

export enum ThemeEnum {
	THEME = 'SET_THEME'
}

export enum AmountEnum {
	AMOUNT = 'SET_AMOUNT'
}

export enum SourceEnum {
	NETWORK = 'SET_SOURCE_NETWORK',
	TOKEN = 'SET_SOURCE_TOKEN'
}

export enum DestinationEnum {
	WALLET = 'SET_DESTINATION_WALLET',
	NETWORK = 'SET_DESTINATION_NETWORK',
	TOKEN = 'SET_DESTINATION_TOKEN',
	ADDRESS = 'SET_DESTINATION_ADDRESS',
	AMOUNT = 'SET_DESTINATION_AMOUNT',
	MEMO = 'SET_DESTINATION_MEMO'
}

export enum KycEnum {
	STATUS = 'SET_KYC_STATUS'
}

export enum KycStatusEnum {
	PROCESS = 'PROCESS',
	INITIAL = 'INITIAL',
	REVIEW = 'REVIEW',
	REFUSED = 'REFUSED',
	PASS = 'PASS',
	DISABLE = 'DISABLE',
	REJECT = 'REJECT'
}

export enum KycL2Enum {
	STATUS = 'SET_KYCL2_STATUS'
}

export enum KycL2StatusEnum {
	INITIAL = 0,
	PENDING = 1,
	PASSED = 2,
	REJECTED = 9
}

export enum KycL2BusinessEnum {
	STATUS = 'SET_KYCL2_Business_STATUS',
	REPR = 'SET_KYCL2_Business_REPR'
}

export enum KycL2BusinessStatusEnum {
	INITIAL = 0,
	PENDING = 1,
	PASSED = 2,
	BASIC = 3,
	REJECTED = 9
}

export enum KycL2BusinessReprEnum {
	NATURAL = 0,
	LEGAL = 1
}

export enum BasicStatusEnum {
	INITIAL = 'INITIAL',
	PROCESS = 'PROCESS',
	REVIEW = 'REVIEW',
	PASS = 'PASS',
	REJECT = 'REJECT',
	SKIP = 'SKIP'
}

export enum ButtonEnum {
	BUTTON = 'SET_BUTTON_STATUS'
}

export enum ProductIdEnum {
	PRODUCTID = 'PRODUCTID'
}

export enum PairEnum {
	PAIR = 'PAIR'
}

export enum DefaultSelectEnum {
	WALlET = 'Select Wallet',
	TOKEN = 'Select Token',
	NETWORK = 'Select Network'
}

export enum AvailableCurrenciesEnum {
	SET = 'SET'
}

type SourceNetworks = 'ETH' | 'GLMR' | DefaultSelectEnum.NETWORK;

type VerificationAction = {
	type: VerificationEnum;
	payload: boolean | string;
};

type KycAction = {
	type: KycEnum;
	payload: KycStatusEnum;
};

type KycL2Action = {
	type: KycL2Enum;
	payload: KycL2StatusEnum;
};

type KycL2BusinessAction = {
	type: KycL2BusinessEnum;
	payload: number;
};

type ButtonAction = {
	type: ButtonEnum;
	payload: { color: string; text: string };
};

type ThemeAction = {
	type: ThemeEnum;
	payload: Theme;
};

type SourceAction = {
	type: SourceEnum;
	payload: string;
};

type DestinationAction = {
	type: DestinationEnum;
	payload: string;
};

type AmountAction = {
	type: AmountEnum;
	payload: string;
};

type ProductIdAction = {
	type: ProductIdEnum;
	payload: string;
};

type PairAction = {
	type: PairEnum;
	payload: string;
};

type AvailableCurrenciesAction = {
	type: AvailableCurrenciesEnum;
	payload: { sourceNetworks: any; destinationNetworks: any };
};

type Action =
	| VerificationAction
	| ButtonAction
	| KycAction
	| KycL2Action
	| KycL2BusinessAction
	| ThemeAction
	| SourceAction
	| DestinationAction
	| AmountAction
	| ProductIdAction
	| PairAction
	| AvailableCurrenciesAction;

type State = {
	isUserVerified: boolean;
	account: string;
	isNetworkConnected: boolean;
	kycStatus: KycStatusEnum;
	kycL2Status: KycL2StatusEnum;
	kycL2Business: KycL2BusinessStatusEnum | null;
	kycL2BusinessRepr: KycL2BusinessReprEnum | null;
	accessToken: string;
	refreshToken: string;
	buttonStatus: { color: string; text: string };
	theme: Theme;
	sourceNetwork: string;
	sourceToken: string;
	destinationWallet: string;
	destinationNetwork: string;
	destinationToken: string;
	destinationAddress: string;
	destinationAmount: string;
	destinationMemo: string;
	amount: string;
	productId: string;
	pair: string;
	availableSourceNetworks: any | null;
	availableDestinationNetworks: any | null;
};

enum ButtonName {
	CONNECT_WALLET = 'CONNECT_WALLET',
	CHANGE_NETWORK = 'CHANGE_NETWORK',
	PASS_KYC = 'PASS_KYC',
	CHECK_KYC = 'CHECK_KYC',
	CHECK_KYC_L2 = 'CHECK_KYC_L2',
	PASS_KYC_L2 = 'PASS_KYC_L2',
	LOGIN = 'LOGIN'
}

type ButtonStatus = { [key in ButtonName]: { color: ColorType; text: string } };

export const button: ButtonStatus = {
	CONNECT_WALLET: { color: 'default', text: 'Connect Wallet' },
	CHANGE_NETWORK: { color: 'error', text: 'Change Network' },
	PASS_KYC: { color: 'warning', text: 'Pass KYC L1' },
	CHECK_KYC: { color: 'success', text: 'Check KYC L1' },
	PASS_KYC_L2: { color: 'warning', text: 'Pass KYC L2' },
	CHECK_KYC_L2: { color: 'warning', text: 'Check KYC L2' },
	LOGIN: { color: 'default', text: 'Login' }
};

const initialState: State = {
	isUserVerified: false,
	account: '',
	isNetworkConnected: false,
	accessToken: '',
	refreshToken: '',
	kycStatus: KycStatusEnum.PROCESS,
	kycL2Status: KycL2StatusEnum.INITIAL,
	kycL2Business: null,
	kycL2BusinessRepr: null,
	buttonStatus: button.CONNECT_WALLET,
	theme: darkTheme,
	destinationWallet: DefaultSelectEnum.WALlET,
	sourceNetwork: DefaultSelectEnum.NETWORK as SourceNetworks,
	sourceToken: DefaultSelectEnum.TOKEN,
	destinationNetwork: DefaultSelectEnum.NETWORK,
	destinationToken: DefaultSelectEnum.TOKEN,
	destinationAddress: '',
	destinationAmount: '',
	destinationMemo: '',
	amount: '',
	productId: '',
	pair: '',
	availableSourceNetworks: null,
	availableDestinationNetworks: null
};

type Dispatch = (action: Action) => void;

const AuthContext = createContext<{ state: State; dispatch: Dispatch } | undefined>(undefined);

const authReducer = (state: State, action: Action): State => {
	switch (action.type) {
		case VerificationEnum.ACCOUNT:
			return { ...state, account: action.payload as string };
		case VerificationEnum.NETWORK:
			return { ...state, isNetworkConnected: action.payload as boolean };
		case VerificationEnum.USER:
			return { ...state, isUserVerified: action.payload as boolean };
		case VerificationEnum.ACCESS:
			return { ...state, accessToken: action.payload as string };
		case VerificationEnum.REFRESH:
			return { ...state, refreshToken: action.payload as string };
		case KycEnum.STATUS:
			return { ...state, kycStatus: action.payload };
		case KycL2Enum.STATUS:
			return { ...state, kycL2Status: action.payload };
		case KycL2BusinessEnum.STATUS:
			return { ...state, kycL2Business: action.payload };
		case KycL2BusinessEnum.REPR:
			return { ...state, kycL2BusinessRepr: action.payload };
		case ButtonEnum.BUTTON:
			return { ...state, buttonStatus: action.payload };
		case ThemeEnum.THEME:
			return { ...state, theme: action.payload };
		case AmountEnum.AMOUNT:
			return { ...state, amount: action.payload };
		case SourceEnum.NETWORK:
			return { ...state, sourceNetwork: action.payload as SourceNetworks };
		case SourceEnum.TOKEN:
			return { ...state, sourceToken: action.payload };
		case DestinationEnum.WALLET:
			return { ...state, destinationWallet: action.payload };
		case DestinationEnum.NETWORK:
			return { ...state, destinationNetwork: action.payload };
		case DestinationEnum.TOKEN:
			return { ...state, destinationToken: action.payload };
		case DestinationEnum.ADDRESS:
			return { ...state, destinationAddress: action.payload };
		case DestinationEnum.AMOUNT:
			return { ...state, destinationAmount: action.payload };
		case DestinationEnum.MEMO:
			return { ...state, destinationMemo: action.payload };
		case ProductIdEnum.PRODUCTID:
			return { ...state, productId: action.payload };
		case PairEnum.PAIR:
			return { ...state, pair: action.payload };
		case AvailableCurrenciesEnum.SET:
			return { 
				...state,
				availableSourceNetworks: action.payload.sourceNetworks,
				availableDestinationNetworks: action.payload.destinationNetworks
			};
		default:
			return state;
	}
};

export const AuthProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
	const [ state, dispatch ] = useReducer(authReducer, initialState);
	const value = { state, dispatch };
	const { account, isNetworkConnected, kycStatus, kycL2Status } = state;

	useEffect(() => {
		if (!account) {
			dispatch({
				type: ButtonEnum.BUTTON,
				payload: button.CONNECT_WALLET
			});
		}

		if (!isNetworkConnected && account) {
			dispatch({
				type: ButtonEnum.BUTTON,
				payload: button.CHANGE_NETWORK
			});
		}

		if (account && isNetworkConnected) {
			dispatch({ type: ButtonEnum.BUTTON, payload: button.LOGIN });
		}

		if (kycStatus === KycStatusEnum.PASS && kycL2Status === KycL2StatusEnum.PASSED && account && isNetworkConnected) {
			dispatch({ type: VerificationEnum.USER, payload: true });
		} else {
			dispatch({ type: VerificationEnum.USER, payload: false });

		}

		// TODO: please do not delete comment section below ;)
		// if (( kycStatus !== KycStatusEnum.PASS && kycL2Status !== KycL2StatusEnum.PASSED ) || !account || !isNetworkConnected) {
		// 	dispatch({ type: VerificationEnum.USER, payload: false });
		// } else if (
		// 	kycStatus === KycStatusEnum.PASS &&
		// 	isNetworkConnected &&
		// 	account &&
		// 	kycL2Status === KycL2StatusEnum.PASSED
		// ) {
		// 	dispatch({ type: VerificationEnum.USER, payload: true });
		// }
	}, [ account, isNetworkConnected, kycStatus, kycL2Status ]);

	useEffect(() => {
		axios.request({
			url: `${BASE_URL}cex/currencies`
		})
			.then(function(response: any) {
				dispatch({
					type: AvailableCurrenciesEnum.SET,
					payload: response.data
				});
			})
			.catch(function(response: any) {
				console.log(response);
				console.error();
			});

	}, []);

	return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
};

export const useStore = () => {
	const context = useContext(AuthContext);
	if (context === undefined) {
		throw new Error('useStore must be used within a AuthProvider');
	}

	return context;
};
