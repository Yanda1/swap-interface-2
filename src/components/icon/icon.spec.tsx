import { CSSProperties } from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import { AuthProvider } from '../../helpers';
import { Icon } from '../../components';
import type { IconType, SizeType } from '../../components';

describe('Icon', () => {
	it.each<[IconType | undefined, SizeType, CSSProperties]>([
		['moon', 'medium', { padding: 12 }],
		['eth', 'large', { marginLeft: 20, border: '1px solid red' }],
		['glmr', 20, { padding: 12 }]
	])('should match snapshot for values icon: %s, size: %s and style: %s', (icon, size, style) => {
		const { getByTestId } = render(
			<AuthProvider>
				<Icon icon={icon} size={size} style={style} />
			</AuthProvider>
		);

		expect(getByTestId('icon')).toMatchSnapshot();
	});

	it('should return default icon if icon is not defined', () => {
		const { getByTestId } = render(
			<AuthProvider>
				<Icon icon={undefined} />
			</AuthProvider>
		);

		expect(getByTestId('icon')).toMatchSnapshot();
	});
});
