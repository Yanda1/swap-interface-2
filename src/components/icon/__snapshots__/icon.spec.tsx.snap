// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match snapshot for values icon: eth, size: large and style: { marginLeft: 20, border: '1px solid red' } 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: unset;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
}

.c0 > * {
  width: 1em;
  height: 1em;
  font-size: 2.5rem;
}

<div
  class="c0"
  data-testid="icon"
  style="margin-left: 20px; border: 1px solid red;"
>
  <svg
    theme="[object Object]"
  >
    eth.svg
  </svg>
</div>
`;

exports[`Icon should match snapshot for values icon: glmr, size: 20 and style: { padding: 12 } 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: unset;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
}

.c0 > * {
  width: 1em;
  height: 1em;
  font-size: 1.25rem;
}

<div
  class="c0"
  data-testid="icon"
  size="20"
  style="padding: 12px;"
>
  <svg
    theme="[object Object]"
  >
    glmr.svg
  </svg>
</div>
`;

exports[`Icon should match snapshot for values icon: moon, size: medium and style: { padding: 12 } 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: unset;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
}

.c0 > * {
  width: 1em;
  height: 1em;
  font-size: 2rem;
}

<div
  class="c0"
  data-testid="icon"
  style="padding: 12px;"
>
  <svg
    theme="[object Object]"
  >
    moon.svg
  </svg>
</div>
`;

exports[`Icon should return default icon if icon is not defined 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: unset;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
}

.c0 > * {
  width: 1em;
  height: 1em;
  font-size: 2rem;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg
    theme="[object Object]"
  >
    black-circle.svg
  </svg>
</div>
`;
